core/lib/types/src/fee_model.rs

```diff
@@ -46,6 +46,30 @@ impl BatchFeeInput {
             fair_pubdata_price,
         })
     }
+
+    pub fn from_protocol_version(
+        protocol_version: Option<ProtocolVersionId>,
+        l1_gas_price: u64,
+        fair_l2_gas_price: u64,
+        fair_pubdata_price: Option<u64>,
+    ) -> Self {
+        protocol_version
+            .filter(|version: &ProtocolVersionId| version.is_post_1_4_1())
+            .map(|_| {
+                Self::PubdataIndependent(PubdataIndependentBatchFeeModelInput {
+                    fair_pubdata_price: fair_pubdata_price
+                        .expect("No fair pubdata price for 1.4.1 miniblock"),
```

Comment:
nit: not alwaays "a miniblock", I'd just leave it as "No fair pubdata price for 1.4.1"